frontends/bsd/src/screens/write_in_adjudication_screen.tsx

```diff
@@ -502,13 +533,15 @@ export function WriteInAdjudicationScreen({
     await onAdjudicationComplete?.(sheetId, side, adjudications);
   }, [adjudications, onAdjudicationComplete, setWriteInPresets, sheetId, side]);
 
-  const selectedContestIdOrDefault = selectedContestId ?? contestIds[0];
-  const selectedContestIndex = contestIds.indexOf(selectedContestIdOrDefault);
-  const isFirstContestSelected = selectedContestIndex === 0;
+  const selectedContestIdOrDefault =
+    selectedContestId ?? contestsWithWriteInsIds[0];
```

Comment:
do you know why we have a default here instead of using a default in the `useState` for `selectedContestId`? It seems once we're done adjudicating all of the contests this component will get unmounted, so it won't need a default anymore

Reply:
I needed a way to make the component automatically move to the first contest of the back page after completing the front, so I set the selected contest ID to `undefined` (L529) so that we can get the first one automatically when the new `allContestIds` prop comes through for the back side. If I had done `const [selectedContestId, setSelectedContestId] = useState(contestsWithWriteInsIds[0])` instead, then I'd have no good way to clear the stale state of the front page's last selected contest when adjudicating the front was complete.

Reply:
Ah interesting. It's not totally clear as the reader, since you can't easily see that the component will be rerendered with a different sheet on completion of adjudication. I would suggest:
- at a minimum adding a comment
- potentially trying to colocate the selected sheet and selected contest state so that the dependency can be seen clearly
- add a key prop to this component in order to reset the state when changing sheets (see https://twitter.com/sebmarkbage/status/1262937670444974081). I think there's a valid argument for saying this is a "master/detail" situation

Reply:
More on that here: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key

Reply:
Thanks for digging into this. Using a `key` is certainly a better approach, which I've applied in 97522ab8c3ac4c65952c97fa500e3eb94ecc664f.

Reply:
Awesome. Yeah I've run into problems like this a few times in Arlo and have been trying to get my head around how to solve them. Still feels tricky for me